app/questionnaire/path_finder.py

```diff
@@ -180,6 +180,10 @@ def get_blocks(self):
             if repeating_rule:
                 no_of_repeats = evaluate_repeat(repeating_rule, self.answer_store)
 
+            if no_of_repeats > 50:
```

Comment:
In my opinion for now a hard coded limit would be okay. When we have the scenario where we need to have varying number of repeats based on question we should look at making it configurable.

It might be good to localise this logic to the evaluate_repeat function